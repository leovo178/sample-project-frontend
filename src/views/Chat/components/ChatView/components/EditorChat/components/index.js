export { default as SendReservationDialog } from './SendReservationDialog'
