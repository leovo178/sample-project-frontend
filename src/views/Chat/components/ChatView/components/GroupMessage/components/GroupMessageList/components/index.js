export { default as ImageMessage } from './ImageMessage'
export { default as LastCardIndicator } from './LastCardIndicator'
export { default as FileMessage } from './FileMessage'
export { default as TextMessage } from './TextMessage'
export { default as ReservationMessage } from './ReservationMessage'
