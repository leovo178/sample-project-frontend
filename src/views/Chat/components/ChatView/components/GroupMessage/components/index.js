export { default as GroupBreak } from './GroupBreak'
export { default as GroupMessageList } from './GroupMessageList'
