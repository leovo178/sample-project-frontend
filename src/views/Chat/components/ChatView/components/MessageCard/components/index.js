export { default as FileMessage } from './FileMessage'
export { default as ImageMessage } from './ImageMessage'
export { default as LastCardIndicator } from './LastCardIndicator'
