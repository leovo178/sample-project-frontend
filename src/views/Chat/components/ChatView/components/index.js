export { default as EditorChat } from './EditorChat'
export { default as GroupMessage } from './GroupMessage'
export { default as Header } from './Header'
export { default as MessageCard } from './MessageCard'
export { default as PickerEmoji } from './PickerEmoji'
export { default as ViewMessage } from './ViewMessage'
export { default as MessageReservation } from './MessageReservation'
