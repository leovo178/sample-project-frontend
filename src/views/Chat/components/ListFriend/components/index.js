// export { default as FriendTag } from './components/FriendTag'
export { default as Header } from './components/Header'
export { default as ListFriendTag } from './components/ListFriendTag'
