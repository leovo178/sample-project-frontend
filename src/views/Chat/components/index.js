export { default as ListFriend } from './components/ListFriend'
export { default as ChatView } from './components/ChatView'
