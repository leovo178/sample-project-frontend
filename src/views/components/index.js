export { default as NavBar } from './NavBar'
export { NavBar as NavBarWithoutRouter } from './NavBar'
export { default as FormEditor } from './FormEditor'
export { default as ModifyDialog } from './ModifyDialog'
export { default as DeleteDialog } from './DeleteDialog'
export { default as Loading } from './Loading'
export { default as LargeTable } from './LargeTable'
export { default as ActionInputBox } from './ActionInputBox'
export { default as CreateInputBox } from './CreateInputBox'
export { default as SuspenseLoading } from './SuspenseLoading'
