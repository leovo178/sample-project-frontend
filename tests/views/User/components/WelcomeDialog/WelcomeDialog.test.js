import React from 'react'
import { cleanup, fireEvent, render, waitFor } from '@testing-library/react'

import { WelcomeDialog } from '@views/User/components'

describe('WelcomeDialog', () => {
	const mockProps = {
		setDialogVisible: jest.fn(),
	}

	afterEach(() => {
		cleanup()
	})

	it('should match snapshot', () => {
		const { container, getByText } = render(<WelcomeDialog {...mockProps} />)

		expect(container).toMatchSnapshot()
		expect(getByText('Welcome to Vatech !!!')).toBeTruthy()
	})

	it('should call setDialogVisible correctly when click Create user button', () => {
		const { getByTestId, getByText } = render(<WelcomeDialog {...mockProps} />)
		const createUserButton = getByTestId('create-user-button')

		fireEvent.click(createUserButton)

		waitFor(() => expect(getByText('Welcome to Vatech !!!')).toBeFalsy())
		expect(mockProps.setDialogVisible).toHaveBeenCalled()
	})
})
