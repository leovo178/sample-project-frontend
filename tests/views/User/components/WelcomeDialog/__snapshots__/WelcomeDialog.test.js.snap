// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeDialog should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-5 makeStyles-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-6"
    >
      <h6
        class="MuiTypography-root makeStyles-welcometext-2 MuiTypography-subtitle1"
      >
        Welcome to Vatech !!!
      </h6>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3 MuiButton-fullWidth"
        data-testid="create-user-button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Create user
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
